src/App.tsx

```diff
@@ -15,36 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+interface PrepareGoodsTypes {
+  sortField: string;
+  reversed: boolean;
+}
+
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reversed }: PrepareGoodsTypes,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  function reset() {
+    setReversed(false);
+    setSortField('');
+  }
```

Comment:
[CODE STYLE] - Function's name should start with a verb (so you could easily tell what your method actually do); Consider renaming the `reset` function to `handleReset` to clearly indicate its purpose and follow best practices.